src/scripts/main.js

```diff
@@ -1,3 +1,417 @@
+      }
+
+      const body = document.querySelector('tbody');
+
+      body.innerHTML = '';
+
+      names.forEach((namee) => {
+        body.appendChild(namee.element);
+      });
+    }
+
+    if (columnName.textContent === 'Position') {
+      clickCounterPosition++;
+
+      const positions = [...document.querySelectorAll('tr')]
+        .map((row) => {
+          return { position: row.children[1].textContent, element: row };
+        })
+        .filter((obj) => obj.position !== 'Position');
+
+      if (clickCounterPosition === 1) {
+        positions.sort((a, b) => a.position.localeCompare(b.position));
+      }
+
+      if (clickCounterPosition === 2) {
+        positions.sort((b, a) => a.position.localeCompare(b.position));
+        setTimeout((a) => (clickCounterPosition = 0), 2000);
+      }
+
+      const body = document.querySelector('tbody');
+
+      body.innerHTML = '';
+
+      positions.forEach((position) => {
+        body.appendChild(position.element);
+      });
+    }
+
+    if (columnName.textContent === 'Office') {
+      clickCounterOffice++;
+
+      const offices = [...document.querySelectorAll('tr')]
+        .map((row) => {
+          return { office: row.children[2].textContent, element: row };
+        })
+        .filter((obj) => obj.office !== 'Office');
+
+      if (clickCounterOffice === 1) {
+        offices.sort((a, b) => a.office.localeCompare(b.office));
+      }
+
+      if (clickCounterOffice === 2) {
+        offices.sort((b, a) => a.office.localeCompare(b.office));
+        setTimeout((a) => (clickCounterOffice = 0), 2000);
+      }
+
+      const body = document.querySelector('tbody');
+
+      body.innerHTML = '';
+
+      offices.forEach((office) => {
+        body.appendChild(office.element);
+      });
+    }
+
+    if (columnName.textContent === 'Age') {
+      clickCounterAge++;
+
+      const ages = [...document.querySelectorAll('tr')]
+        .map((row) => {
+          return { age: row.children[3].textContent, element: row };
+        })
+        .filter((obj) => obj.age !== 'Age');
+
+      if (clickCounterAge === 1) {
+        ages.sort((a, b) => a.age - b.age);
+      }
+
+      if (clickCounterAge === 2) {
+        ages.sort((b, a) => a.age - b.age);
+        setTimeout((a) => (clickCounterAge = 0), 2000);
+      }
+
+      const body = document.querySelector('tbody');
+
+      body.innerHTML = '';
+
+      ages.forEach((age) => {
+        body.appendChild(age.element);
+      });
+    }
+
+    if (columnName.textContent === 'Salary') {
+      clickCounterSalary++;
+
+      const salarys = [...document.querySelectorAll('tr')]
+        .map((row) => {
+          return { salary: row.children[4].textContent, element: row };
+        })
+        .filter((obj) => obj.salary !== 'Salary');
+
+      if (clickCounterSalary === 1) {
+        salarys.sort(
+          (a, b) =>
+            Number(a.salary.slice(1).split(',').join('')) -
+            Number(b.salary.slice(1).split(',').join('')),
```

Comment:
The salary formatting logic may not work correctly if the salary is not formatted as expected. Consider using a more robust method to handle currency formatting.